### ISSUE TITLE
False negative: `unnecessary-negation` misses `not (a is not b)` and `not (a not in b)`

### ISSUE THREAD

**Issue by Synvoya** [PyCQA/pylint#11140]
### Bug description

`unnecessary-negation` (C0117) does not flag `not (a is not b)` or `not (a not in b)`, even though both are trivially simplifiable:

- `not (a is not b)` ≡ `a is b`
- `not (a not in b)` ≡ `a in b`

The `is` / `in` cases **are** flagged; only their siblings `is not` / `not in` are silently missed.

```python
def f(a, b):
    if not a is not b:        # not flagged — should suggest `a is b`
        pass
    if not a not in [1, 2]:   # not flagged — should suggest `a in [1, 2]`
        pass
```

### Command used

```shell
pylint --disable=all --enable=unnecessary-negation f.py
```

### Pylint output

Nothing for the two cases above (the `is` / `in` variants without the second negation are correctly flagged).

### Expected behavior

Both should be flagged, consistent with the existing `is` / `in` handling.

### Root cause

`pylint/checkers/refactoring/not_checker.py`: the `reverse_op` inversion table has `is` and `in` as keys but omits `is not` and `not in`, so the checker early-returns (`if operator not in self.reverse_op`) for those two operators.

### Pylint version

```
pylint 4.x (main branch), astroid 4.x
```
